test/e2e/step_plugin_test.go

```diff
@@ -0,0 +1,327 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/tj/assert"
+	"gopkg.in/yaml.v2"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/test/fixtures"
+	"github.com/argoproj/argo-rollouts/utils/plugin/types"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const E2EStepPluginName = "step/e2e-test"
+const E2EStepPluginNameDisabled = "step/e2e-test-disabled"
+
+type StepPluginSuite struct {
+	fixtures.E2ESuite
+}
+
+func TestStepPluginSuite(t *testing.T) {
+	suite.Run(t, new(StepPluginSuite))
+}
+
+func (s *StepPluginSuite) SetupSuite() {
+	s.E2ESuite.SetupSuite()
+	if !IsStepPluginConfigured(&s.Common, s.GetControllerConfig()) {
+		s.T().SkipNow()
```

Comment:
Should I configure E2E by default with a valid step plugin

Reply:
I think we should but can do it, in a different PR.